propagators/opentelemetry-propagator-instana/src/InstanaPropagator.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Context,
+  trace,
+  isSpanContextValid,
+  isValidSpanId,
+  isValidTraceId,
+  TextMapGetter,
+  TextMapPropagator,
+  TextMapSetter,
+  TraceFlags,
+  INVALID_TRACEID,
+  INVALID_SPANID,
+} from '@opentelemetry/api';
+
+/** Instana header names */
+export const INSTANA_TRACE_ID_HEADER = 'X-INSTANA-T';
+export const INSTANA_SPAN_ID_HEADER = 'X-INSTANA-S';
+export const INSTANA_LEVEL_HEADER = 'X-INSTANA-L';
+
+const FIELDS = [
+  INSTANA_TRACE_ID_HEADER,
+  INSTANA_SPAN_ID_HEADER,
+  INSTANA_LEVEL_HEADER,
+];
+
+function readHeader(
+  carrier: unknown,
+  getter: TextMapGetter,
+  key: string
+): string {
+  let header = getter.get(carrier, key.toLowerCase());
```

Comment:
Why are headers sent uppercase but read lowercase?
For http/http2 module this works fine as these servers normalize to lowercase. But there are other protocols where headers are case sensitive or at least no normalization is done (I think kafka and/or grpc).

Usually I would expect that extract after inject like this extracts what was injected:
```js
const carrier = {}
propagation.inject(ctx, carrier)
const ctx1 = propagation.extract(ROOT_CONTEXT, carrier)
```

What about sending them already lowercase like the W3C headers?

Reply:
> Why are headers sent uppercase but read lowercase? 
> For http/http2 module this works fine as these servers normalize to lowercase.

If we read them upper case, it won't work for http, exactly because of that normalization being done by Node.js core http modules. I only realized that when trying it out in an actual app. In hindsight that should have been obvious though.

> What about sending them already lowercase like the W3C headers?

As for sending, by convention those headers are sent as all upper case by all Instana tracers. Sure enough, it shouldn't make a difference as everybody reading them should read them case insensitive, but I still would like to go with the conventions for these headers that have been established in Instana basically since forever.

 > But there are other protocols where headers are case sensitive or at least no normalization is done (I think kafka and/or grpc).

That is a very good point, thanks for bringing it up. I really should not read them lower case, but instead read them case insensitive, even if there is no normalization in place. I'll fix that.
 
> Usually I would expect that extract after inject like this extracts what was injected:
> 
> ```js
> const carrier = {}
> propagation.inject(ctx, carrier)
> const ctx1 = propagation.extract(ROOT_CONTEXT, carrier)
> ```

Well, the asymmetrical behavior here is introduced by the convention for all upper case for these headers that have been established elsewhere and Node.js normalization to lower case. I'll make the reading side case insensitive independent of the protocol and the sending side will stick to all upper case.

Reply:
Please prefer fixup commits instead force pushing as it's by far easier to review. Not a big deal this time as change was small and isolated.
During integration all your commits are anyway squashed.